cds_ils/importer/api.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+import click
+from celery import shared_task
+from invenio_app_ils.errors import IlsValidationError
+from invenio_db import db
+
+from cds_ils.importer.XMLRecordDump import XMLRecordDump
+from cds_ils.importer.XMLRecordLoader import XMLRecordDumpLoader
+from cds_ils.importer.parse_xml import get_records_list
+
+records_logger = logging.getLogger("records_errored")
+
+
+@shared_task()
+def process_dump(data, provider, source_type):
+    recorddump = XMLRecordDump(
+        data,
+        source_type=source_type,
+    )
+    try:
+        report = XMLRecordDumpLoader.process(recorddump, provider)
+        db.session.commit()
+        return report
+    except Exception as e:
+        db.session.rollback()
+        raise e
+
+
+def import_record(data, provider, source_type=None, eager=False):
+    """Import record from dump."""
+
+    source_type = source_type or 'marcxml'
+    assert source_type in ['marcxml', 'json']
+
+    if eager:
+        return process_dump(data, provider, source_type=source_type)
+    else:
+        process_dump.delay(data, provider, source_type=source_type)
+
+
+def import_from_xml(sources, source_type, provider, eager=True):
+    """Load xml file."""
+
+    for idx, source in enumerate(sources, 1):
+        click.echo(
```

Comment:
in `api.py`, you don't know that you are in CLI. I would suggest to remove these `click.echo` and replace them with a configurable logger printing to console instead (if not already the case by default in Invenio).
Same for all other files.
In the end, all this should be called via a UI form, and output displayed in a web page.

If you want to keep the colors, you might have to pass a `logger_func` in the params, which corresponds to `click.echo` when called from the CLI.

Reply:
I left it for later, there is a separate task for logging in the importer. I had in mind to do it similarly as the reports for the user sync. WDYT?